### DISCUSSION TITLE
Bundle size increased by 1mb+ in latest version

### DISCUSSION THREAD

**Discussion by 0xRapid** [rainbow-me/rainbowkit#1719] [General]
Did anyone else experience that? I also can't seem to be picky about those components which means it comes as the base bundle. Any clue? Noticed it when upgrading from 1.0.7 to 1.3.2. I believe it has to do something with the locale.

**Comment by 0xRapid** [comment]
![image](https://github.com/rainbow-me/rainbowkit/assets/14364861/f5a0d3aa-92fe-4b2c-8236-fc4417459c70)
and the list of files goes on there.

If I choose locale as en by default, there's no treeshake or anything to make sure it doesn't add that additional 1mb almost? :(

**Comment by o-az** [comment]
> Did anyone else experience that? I also can't seem to be picky about those components which means it comes as the base bundle. Any clue? Noticed it when upgrading from 1.0.7 to 1.3.2. I believe it has to do something with the locale.

looks like a bug to me
https://packagephobia.com/result?p=%40rainbow-me%2Frainbowkit
https://pkg-size.dev/@rainbow-me/rainbowkit

**Reply by 0xRapid** [reply]
Oh wow, that increase in size. Almost doubled in that version :O

**Reply by 0xRapid** [reply]
![image](https://github.com/rainbow-me/rainbowkit/assets/14364861/094c3045-b77d-4dd5-a0d4-c1d59e290c67)

**Reply by o-az** [reply]
> Oh wow, that increase in size. Almost doubled in that version :O

What you should really be looking at is your app build output in both versions and comparing the sizes because the not everything in the package is bundled

**Comment by 0xmkh** [comment]
@0xRapid We do tree shake the locales [here](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/locales/index.ts#L93-L103). The reason why the package has become much larger in size is because we've introduced internalisation with [i18n-js](https://www.npmjs.com/package/i18n-js).

**Comment by 0xRapid** [comment]
> @0xRapid We do tree shake the locales [here](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/locales/index.ts#L93-L103). The reason why the package has become much larger in size is because we've introduced internalisation with [i18n-js](https://www.npmjs.com/package/i18n-js).

What if I am not seeking to use any internalization in my app? I just don't want to increase my bundle size by 1mb just because of that, usually those stuff are loaded dynamically and does not increase the base bundle size like it did here. 

Is there a solution for that? @KosmosKey

**Reply by 0xmkh** [reply]
In that case you wanna be looking at the bundle size in your app instead of looking at the size of the whole package itself. We don't bundle all languages at once only if you choose to, but we're only including english language at first and bundle other ones when they are selected.

Yes the bundle might increase a bit, but that's because we wanna make sure rainbowkit is being used by other folks who are non english speakers too.

Here is an example [link](https://bundlephobia.com/package/@rainbow-me/rainbowkit@1.3.3) of bundlephobia

![image](https://github.com/rainbow-me/rainbowkit/assets/53529533/555cb2dd-a88b-4f36-b601-7ac325cfdeac)

**Reply by 0xmkh** [reply]
@0xRapid The best solution for now is to not include the `locale` prop in your `<RainbowKitProvider />` component. This way we'll make sure to only use the english language and not bundle the other languages.

**Reply by 0xRapid** [reply]
Yeah, I am not using `locale` in the example I provided. 

I'm saying it's specifically rainbowkit that makes it increase in 1mb since the change, check the picture in the main post, you can see all the file changes are related to rainbowkit.

I have to say as someone who is going hardcore performance-wise for SEO, adding 1 MB (unzipped) to our site bundle is huge.

I did try and test it with `locale="en"` before I posted here btw, just to see maybe it won't load anything else. But it does :(.  The default did not have any language. 

Basically you are saying since internalisation is now default, I can not avoid decreasing it right? :(

**Reply by 0xmkh** [reply]
@0xRapid Feel free to update to RainbowKit version `1.3.4`. We no longer use external packages to bundle the locales, but rather using our own now. This should be enough for the SEO improvements at least (30%) improvement in bundle size.

![image](https://github.com/rainbow-me/rainbowkit/assets/53529533/ce1598de-ee6f-488c-9c09-b1101e5fb84a)

**Reply by 0xmkh** [reply]
Also one thing to keep in mind to not look at the installation size, but rather bundle size. Those are two different things. If we take `viem` as an example we see that they have 7.59 MB in installation size, but that doesn't mean it will have 7.59 MB in bundle size.

**Reply by 0xRapid** [reply]
Yeah I know brother, trust me I went through alot of pain optimizing our bundles, days of optimizations of imports in a monorepo and many adaptions to get the bundle low size. As you can see in the picture in the main post, I have even used a github action in order to check for changes in bundle sizes, that's how I noticed it. 

Thank you guys for optimizing it. I might contribute later on and see if I can improve some additional stuff in the repo once I find a little bit of time. I appreciate it guys! I'll give it a try in a moment!

**Reply by 0xRapid** [reply]
Getting pretty much the same results, it actually increased from 8.8% increase to 10.1%, I'll try investigating it a bit more on my end to see.
![image](https://github.com/rainbow-me/rainbowkit/assets/14364861/53d6a5fb-0f17-47f4-9ddb-c8c6be6df746)

**Reply by 0xmkh** [reply]
These are our locales file. We could remove all of them and use API fetching instead, but that would not be neccessary unless we have to. Can you please check the bundle size when you load the browser instead of checking the package itself ? 

Not everything is bundled in the package those are just locale files which takes up some space.

**Reply by 0xmkh** [reply]
We can try to also code split our components in the RainbowKit which is the future long term here.

**Reply by 0xRapid** [reply]
Yeah, I think splitting the components would be ideal. Barrel code is very common and many times causes a high impact that many developers do not see if that's what you mean.

However you are right when it comes to the pictures I posted, these are indeed the node_modules files that were added, sorry for that.

It seems like `bundle-analyzer` also shows me some different data, so I'll check it more and update my findings. Thanks @KosmosKey

**Reply by 0xmkh** [reply]
Sounds good. Yeah let me know if you find anything that affects your bundle size, but RainbowKit should load now faster even on 3G it's 1.21s which should be good enough for a great SEO.
